packages/flutter/test/material/menu_anchor_test.dart

```diff
@@ -1235,6 +1252,215 @@ void main() {
+      await tester.pump();
+
+      expect(opened, equals(<TestMenu>[TestMenu.mainMenu1, TestMenu.subMenu11]));
+      expect(closed, equals(<TestMenu>[TestMenu.subMenu11, TestMenu.mainMenu1]));
+      expect(selected, equals(<TestMenu>[TestMenu.subSubMenu112]));
+    }, variant: TargetPlatformVariant(nonApple));
+
+    testWidgets("disabled items don't respond to accelerators", (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Material(
+            child: MenuBar(
+              key: UniqueKey(),
+              controller: controller,
+              children: createTestMenus(
+                onPressed: onPressed,
+                onOpen: onOpen,
+                onClose: onClose,
+                accelerators: true,
+              ),
+            ),
+          ),
+        ),
+      );
+
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.altLeft);
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.keyM, character: '5');
+      await tester.pump();
+      await tester.sendKeyUpEvent(LogicalKeyboardKey.altLeft);
+      await tester.pump();
+
+      expect(opened, isEmpty);
+      expect(closed, isEmpty);
+      expect(selected, isEmpty);
+      // Selecting a non-submenu item should close all the menus.
+      expect(find.text(TestMenu.subMenu00.label), findsNothing);
+    }, variant: TargetPlatformVariant(nonApple));
+
+    testWidgets("Apple platforms don't react to accelerators", (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Material(
+            child: MenuBar(
+              key: UniqueKey(),
+              controller: controller,
+              children: createTestMenus(
+                onPressed: onPressed,
+                onOpen: onOpen,
+                onClose: onClose,
+                accelerators: true,
+              ),
+            ),
+          ),
+        ),
+      );
+
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.altLeft);
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.keyM, character: 'm');
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.keyM, character: 'm');
+      await tester.sendKeyUpEvent(LogicalKeyboardKey.altLeft);
+      await tester.pump();
+
+      expect(opened, isEmpty);
+      expect(closed, isEmpty);
+      expect(selected, isEmpty);
+
+      // Or with the option key equivalents.
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.altLeft);
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.keyM, character: 'µ');
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.keyM, character: 'µ');
+      await tester.sendKeyUpEvent(LogicalKeyboardKey.altLeft);
+      await tester.pump();
+
+      expect(opened, isEmpty);
+      expect(closed, isEmpty);
+      expect(selected, isEmpty);
+    }, variant: const TargetPlatformVariant(apple));
+  });
```

Comment:
A thought I had while reading these tests: What will happen if a TextField is focused and then you try to trigger an accelerator with alt + letter?

Reply:
As long as the text field doesn't have alt+letter defined as a shortcut, it will fire the accelerator.

It is true that the accelerators are fired after other shortcuts that are closer to the `primaryFocus` in the widget tree, which might cause some confustion, although it's uncommon to define an Alt-based shortcut, and if you do, you shouldn't be surprised that it takes precedence. I'll add a note about the precedence of accelerators vs other shortcuts.